x/ethbridge/handler.go

```diff
@@ -67,17 +70,55 @@ func handleMsgCreateEthBridgeClaim(ctx sdk.Context, cdc *codec.Codec,
 	return sdk.Result{Events: ctx.EventManager().Events()}
 }
 
-func processSuccessfulClaim(ctx sdk.Context, bankKeeper bank.Keeper, claim string) sdk.Error {
+func processSuccessfulClaim(ctx sdk.Context, supplyKeeper supply.Keeper, claim string) sdk.Error {
 	oracleClaim, err := types.CreateOracleClaimFromOracleString(claim)
 	if err != nil {
 		return err
 	}
 
 	receiverAddress := oracleClaim.CosmosReceiver
-	_, err = bankKeeper.AddCoins(ctx, receiverAddress, oracleClaim.Amount)
+	err = supplyKeeper.MintCoins(ctx, ModuleName, oracleClaim.Amount)
```

Comment:
The `mint` module should be the only one allowed to mint new coins (i.e have a `ModuleAccount` with `Minter` permissions). This line should call the `mintKeeper.MintCoins(ctx, oracleClaim.Amount)` function. 

The EthBridge module should only be able to store and burn those coins (i.e should only have `Burner` permission). 

Also as a side note: this usually causes token invariance, so I'd recommend you to add the simulation to check this